client/js/shout.js

```diff
@@ -224,31 +224,59 @@ $(function() {
 				}
 			});
 
-		if (!chan.hasClass("channel")) {
-			return;
-		}
-
-		var type = data.msg.type;
-		if (type === "message" || type === "action") {
-			var nicks = chan.find(".users").data("nicks");
-			if (nicks) {
-				var find = nicks.indexOf(from);
-				if (find !== -1 && typeof move === "function") {
-					move(nicks, find, 0);
+		if (chan.hasClass("channel")) {
+			var type = data.msg.type;
+			if (type === "message" || type === "action") {
+				var nicks = chan.find(".users").data("nicks");
+				if (nicks) {
+					var find = nicks.indexOf(from);
+					if (find !== -1 && typeof move === "function") {
+						move(nicks, find, 0);
+					}
 				}
 			}
 		}
+
+		return msg;
+	}
+
+	function renderChannelMessages(data) {
+		var documentFragment = $(document.createDocumentFragment());
+		for (var i = 0; i < data.messages.length; i++) {
+			var msg = renderChatMessage({
+				chan: data.id,
+				msg: data.messages[i]
+			});
+			documentFragment.append(msg);
+		}
+		$("#chan-" + data.id + " .messages").append(documentFragment);
+	}
+
+	socket.on("msg", function(data) {
+		var msg = renderChatMessage(data);
+		var target = "#chan-" + data.chan;
+
+		chat.find(target + " .messages")
+			.append(msg)
+			.trigger("msg", [
+				target,
+				data.msg
+			]);
 	});
 
 	socket.on("more", function(data) {
-		var target = data.chan;
-		var chan = chat
-			.find("#chan-" + target)
-			.find(".messages")
-			.prepend(render("msg", {messages: data.messages}))
-			.end();
+		var documentFragment = $(document.createDocumentFragment());
+		for (var i = 0; i < data.messages.length; i++) {
+			var msg = renderChatMessage({
+				chan: data.chan,
+				msg: data.messages[i]
+			});
+			documentFragment.append(msg);
+		}
```

Comment:
You can't directly replace it because first one uses `data.id` the other one uses `data.chan`. This probably needs to be fixed on the server side for consistency though.